crates/loam-cli/src/commands/dev/mod.rs

```diff
@@ -0,0 +1,103 @@
+use clap::Parser;
+use notify::{RecursiveMode, Watcher};
+use std::path::{Path, PathBuf};
+use tokio::time;
+
+use crate::commands::build;
+
+use super::build::build_clients::LoamEnv;
+
+#[derive(Parser, Debug, Clone)]
+#[group(skip)]
+pub struct Cmd {
+    #[command(flatten)]
+    pub build_cmd: build::Cmd,
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Watcher(#[from] notify::Error),
+    #[error(transparent)]
+    Build(#[from] build::Error),
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<(), Error> {
+        // Set LOAM_ENV to development
+        std::env::set_var("LOAM_ENV", "development");
+        let (tx, mut rx) = tokio::sync::mpsc::channel(100);
+        let tx_clone = tx.clone();
+        let mut watcher =
+            notify::recommended_watcher(move |res: Result<notify::Event, notify::Error>| {
+                if let Ok(event) = res {
+                    if let notify::EventKind::Modify(_) = event.kind {
+                        if let Some(path) = event.paths.first() {
+                            println!("File modified: {:?}", path);
+                            // Send a signal through the channel to trigger a rebuild
+                            if let Err(e) = tx_clone.blocking_send(()) {
+                                eprintln!("Error sending through channel: {}", e);
+                            }
+                        }
+                    }
+                }
+            })
+            .unwrap();
+
+        let workspace_root = self
+            .build_cmd
+            .manifest_path
+            .parent()
+            .map(|path| path.to_path_buf())
+            .unwrap_or_else(|| PathBuf::from("."));
```

Comment:
You should be able to do `Path::new(".")` and avoid needing the map. `workspace_root` can be a `Path`

Reply:
got it